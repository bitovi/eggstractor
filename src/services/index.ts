export * from './collection.service';
export * from './token.service';
export * from './variable.service'; 