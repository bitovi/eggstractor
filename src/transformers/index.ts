export * from "./scss.transformer";
export * from "./css.transformer";
