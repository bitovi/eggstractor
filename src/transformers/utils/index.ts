export * from './deduplicate-messages.utils';
export * from './get-style-property-and-value.utils';
export * from './group-by.utils';
export * from './update-padding-based-on-border.utils';
