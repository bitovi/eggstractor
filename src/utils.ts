export default {
  toBase64(str: string): string {
    const base64chars = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/';
    const utf8str = encodeURIComponent(str).replace(/%([0-9A-F]{2})/g,
      (_, p1) => String.fromCharCode(parseInt(p1, 16))
    );
    let i = 0;
    let result = '';
    while (i < utf8str.length) {
      const char1 = utf8str.charCodeAt(i++);
      const char2 = i < utf8str.length ? utf8str.charCodeAt(i++) : NaN;
      const char3 = i < utf8str.length ? utf8str.charCodeAt(i++) : NaN;
      const enc1 = char1 >> 2;
      const enc2 = ((char1 & 3) << 4) | (char2 >> 4);
      const enc3 = ((char2 & 15) << 2) | (char3 >> 6);
      const enc4 = char3 & 63;
      result += base64chars[enc1] + base64chars[enc2] +
        (isNaN(char2) ? '=' : base64chars[enc3]) +
        (isNaN(char3) ? '=' : base64chars[enc4]);
    }
    return result;
  },
  sanitizeName(name: string): string {
    return name.toLowerCase().replace(/[^a-z0-9]/g, '-');
  },
  sanitizeSegment(segment: string): string {
    return segment.toLowerCase().replace(/[^a-z0-9]/g, '-');
  },
  rgbToHex(r: number, g: number, b: number): string {
    const rHex = Math.round(r * 255).toString(16).padStart(2, '0');
    const gHex = Math.round(g * 255).toString(16).padStart(2, '0');
    const bHex = Math.round(b * 255).toString(16).padStart(2, '0');
    return '#' + rHex + gHex + bHex;
  },
  rgbaToString(r: number, g: number, b: number, a: number): string {
    const rHex = Math.round(r * 255).toString(16).padStart(2, '0');
    const gHex = Math.round(g * 255).toString(16).padStart(2, '0');
    const bHex = Math.round(b * 255).toString(16).padStart(2, '0');
    const aHex = Math.round(a * 255).toString(16).padStart(2, '0');
    
    return a === 1 ? 
      '#' + rHex + gHex + bHex : 
      '#' + rHex + gHex + bHex + aHex;
  },
  groupBy<T>(arr: T[], key: (item: T) => string): Record<string, T[]> {
    return arr.reduce((groups, item) => {
      const k = key(item);
      if (!groups[k]) groups[k] = [];
      groups[k].push(item);
      return groups;
    }, {} as Record<string, T[]>);
  }
};