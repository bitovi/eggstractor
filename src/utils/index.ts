export * from './color.utils';
export * from './create-naming-convention'
export * from './constants.utils';
export * from './delay.utils';
export * from './get-valid-stylesheet-format.utils';
export * from './node.utils';
export * from './string.utils';
export * from './test.utils';
export * from './units.utils';
export * from './value.utils';
