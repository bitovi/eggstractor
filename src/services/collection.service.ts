import { ComponentSetToken, ComponentToken, TokenCollection } from '../types';
import { getProcessorsForNode } from '../processors';
import { extractComponentSetToken, extractComponentToken, extractNodeToken } from '../services';
import { getNodePathNames } from '../utils/node.utils';

function getFlattenedValidNodes(node: BaseNode): BaseNode[] {
  const result: BaseNode[] = [];

  function traverse(currentNode: BaseNode) {
    // Skip VECTOR and INSTANCE nodes entirely. INSTANCE nodes which are
    // instances of components, which are not relevant for token extraction.
    if ('type' in currentNode && ['VECTOR', 'INSTANCE'].includes(currentNode.type)) {
      return;
    }

    result.push(currentNode);

    if ('children' in currentNode) {
      for (const child of currentNode.children) {
        traverse(child);
      }
    }
  }

  traverse(node);
  return result;
}

export async function collectTokens(onProgress: (progress: number, message: string) => void) {
  const collection: TokenCollection = { tokens: [], components: {}, componentSets: {} };

  let componentToken: ComponentToken | null = null;
  let componentSetToken: ComponentSetToken | null = null;

  let totalNodes = 0;
  let processedNodes = 0;

  let lastTimestamp = Date.now();
  let lastPercentage = -1;

  async function processNode(node: BaseNode) {
    processedNodes++;

    // calculate integer percent in your 10–95 range
    const currentPercentage = Math.floor((processedNodes / totalNodes) * 85) + 10;
    const shouldUpdate = currentPercentage !== lastPercentage || processedNodes === totalNodes;

    if (shouldUpdate) {
      lastTimestamp = currentPercentage;
      onProgress(currentPercentage, `Processing nodes… ${processedNodes}/${totalNodes}`);

      // throttle yields to at most once every 200 ms
      const now = Date.now();
      if (now - lastTimestamp >= 200) {
        await new Promise((r) => setTimeout(r, 0));
        lastTimestamp = now;
      }
    }

    if ('type' in node && 'boundVariables' in node) {
      if (node.type === 'COMPONENT_SET') {
        componentSetToken = extractComponentSetToken(node);
        collection.componentSets[node.id] = componentSetToken;
      }

      if (node.type === 'COMPONENT') {
        componentToken = extractComponentToken(node, componentSetToken!);
        collection.components[node.id] = componentToken;
      }

      const nodePathNames = getNodePathNames(node);
      const processors = getProcessorsForNode(node);

      for (const processor of processors) {
        const tokens = await extractNodeToken(
          node,
          processor,
          nodePathNames,
          componentToken,
          componentSetToken,
        );
        collection.tokens.push(...tokens);
      }
    }
  }

  onProgress(0, 'Loading pages...');
  await figma.loadAllPagesAsync();

  onProgress(5, 'Counting nodes...');
  const validNodes = figma.root.children.flatMap((page) => getFlattenedValidNodes(page));
  totalNodes = validNodes.length;

  onProgress(10, `Processing ${totalNodes} nodes...`);

  // Process all pages in parallel for maximum speed
  const pagePromises = figma.root.children.map(async (page) => {
    const validNodes = getFlattenedValidNodes(page);

    for (const node of validNodes) {
      await processNode(node);
    }
  });

  await Promise.all(pagePromises);

  return collection as Readonly<TokenCollection>;
}
