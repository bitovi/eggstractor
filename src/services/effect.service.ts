import { VariableToken, TokenCollection } from '../types';
import { sanitizeName } from '../utils';

/**
 * Convert a Figma EffectStyle to a VariableToken
 */
function createEffectStyleToken(effectStyle: EffectStyle): VariableToken | null {
  try {
    // Convert effects array to CSS box-shadow value
    const boxShadowValues = effectStyle.effects
      .filter((effect) => effect.type === 'DROP_SHADOW' || effect.type === 'INNER_SHADOW')
      .map((effect) => {
        const { offset, radius, color, spread } = effect;
        const isInset = effect.type === 'INNER_SHADOW';

        // Convert RGBA to CSS rgba()
        const colorValue = `rgba(${Math.round(color.r * 255)}, ${Math.round(color.g * 255)}, ${Math.round(color.b * 255)}, ${color.a ?? 1})`;

        // Build box-shadow value: [inset] offset-x offset-y blur-radius [spread-radius] color
        const parts = [];
        if (isInset) parts.push('inset');
        parts.push(`${offset?.x || 0}px`);
        parts.push(`${offset?.y || 0}px`);
        parts.push(`${radius || 0}px`);
        if (spread) parts.push(`${spread}px`);
        parts.push(colorValue);

        return parts.join(' ');
      });

    if (boxShadowValues.length === 0) {
      return null;
    }

    const rawValue = boxShadowValues.join(', ');

    return {
      type: 'variable',
      path: [{ name: effectStyle.name, type: 'FRAME' }],
      property: 'box-shadow',
      name: sanitizeName(effectStyle.name),
      value: `$${sanitizeName(effectStyle.name)}`,
      rawValue,
      valueType: null,
      metadata: {
        figmaId: effectStyle.id,
        variableName: effectStyle.name,
        variableTokenType: 'primitive',
      },
    };
  } catch (error) {
    console.warn(`Failed to create token for effect style ${effectStyle.name}:`, error);
    return null;
  }
}

/**
 * Collect all Figma Effect Styles (box-shadow, etc.)
 */
export async function collectAllFigmaEffectStyles(
  collection: TokenCollection,
  onProgress: (progress: number, message: string) => void,
) {
  try {
    onProgress(7, 'Collecting Figma Effect Styles...');
    const effectTokens: VariableToken[] = [];

    // Get all local effect styles
    const effectStyles = await figma.getLocalEffectStylesAsync();

    for (const effectStyle of effectStyles) {
      onProgress(8, `Processing effect style: ${effectStyle.name}`);

      const token = createEffectStyleToken(effectStyle);
      if (token) {
        effectTokens.push(token);
      }
    }

    console.log(`âœ¨ Collected ${effectTokens.length} effect style tokens`);
    collection.tokens.push(...effectTokens);
  } catch (error) {
    console.warn('Failed to collect Figma Effect Styles:', error);
  }
}
