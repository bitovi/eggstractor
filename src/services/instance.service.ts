import { StyleToken, InstanceToken, ComponentToken, TokenCollection } from '../types';
import { getProcessorsForNode } from '../processors';
import { extractNodeToken } from './token.service';
import { getNodePathNames } from '../utils/node.utils';

export interface InstanceOverride {
  instanceId: string;
  instanceName: string;
  componentId?: string;
  overrideTokens: StyleToken[];
  allTokens: StyleToken[];
}

/**
 * Detects and extracts override styles for component instances by comparing
 * fresh style extractions from both instance and component nodes.
 */
export async function detectInstanceOverrides(
  collection: TokenCollection
): Promise<InstanceOverride[]> {
  const overrides: InstanceOverride[] = [];

  for (const [instanceId, instanceToken] of Object.entries(collection.instances)) {
    if (instanceToken.remote || !instanceToken.componentNode) {
      continue; // Skip remote instances and instances without component references
    }

    try {
      // Get the actual Figma nodes for comparison
      const instanceNode = await figma.getNodeByIdAsync(instanceId) as InstanceNode;
      const componentNode = instanceToken.componentNode;
      
      if (!instanceNode || !componentNode) {
        continue;
      }

      // Extract fresh styles from both nodes for accurate comparison
      const componentSetToken = instanceToken.componentNode?.parent?.type === 'COMPONENT_SET' 
        ? collection.componentSets[instanceToken.componentNode.parent.id]
        : undefined;
      
      const componentTokenData = collection.components[componentNode.id];
      
      const instanceStyles = await extractInstanceStyles(instanceNode, componentTokenData, componentSetToken);
      const componentStyles = await extractComponentStyles(componentNode, componentTokenData, componentSetToken);

      // Find overrides by comparing properties
      const overrideTokens: StyleToken[] = [];
      
      for (const instanceStyle of instanceStyles) {
        const correspondingComponentStyle = componentStyles.find(
          componentStyle => componentStyle.property === instanceStyle.property
        );

        // If there's no corresponding token in the component, or values differ, it's an override
        if (!correspondingComponentStyle || 
            correspondingComponentStyle.value !== instanceStyle.value ||
            correspondingComponentStyle.rawValue !== instanceStyle.rawValue) {
          overrideTokens.push(instanceStyle);
        }
      }

      // Include this instance even if there are no overrides (as per requirement)
      overrides.push({
        instanceId,
        instanceName: instanceToken.name,
        componentId: componentNode.id,
        overrideTokens,
        allTokens: instanceStyles,
      });

    } catch (error) {
      console.warn(`Failed to process instance ${instanceToken.name}:`, error);
      // Still include the instance with empty overrides
      overrides.push({
        instanceId,
        instanceName: instanceToken.name,
        componentId: instanceToken.componentNode?.id,
        overrideTokens: [],
        allTokens: [],
      });
    }
  }

  return overrides;
}

/**
 * Extracts all style properties for an instance node by processing it directly.
 * This is used to get fresh token data for comparison.
 */
export async function extractInstanceStyles(
  instanceNode: InstanceNode,
  componentToken?: ComponentToken,
  componentSetToken?: any
): Promise<StyleToken[]> {
  const nodePathNames = getNodePathNames(instanceNode);
  const processors = getProcessorsForNode(instanceNode);
  const allTokens: StyleToken[] = [];

  for (const processor of processors) {
    const tokens = await extractNodeToken(
      instanceNode,
      processor,
      nodePathNames,
      componentToken,
      componentSetToken,
    );

    const styleTokens = tokens.filter((token): token is StyleToken => token.type === 'style');
    allTokens.push(...styleTokens);
  }

  return allTokens;
}

/**
 * Extracts all style properties for a component node by processing it directly.
 */
export async function extractComponentStyles(
  componentNode: ComponentNode,
  componentToken?: ComponentToken,
  componentSetToken?: any
): Promise<StyleToken[]> {
  const nodePathNames = getNodePathNames(componentNode);
  const processors = getProcessorsForNode(componentNode);
  const allTokens: StyleToken[] = [];

  for (const processor of processors) {
    const tokens = await extractNodeToken(
      componentNode,
      processor,
      nodePathNames,
      componentToken,
      componentSetToken,
    );

    const styleTokens = tokens.filter((token): token is StyleToken => token.type === 'style');
    allTokens.push(...styleTokens);
  }

  return allTokens;
}
