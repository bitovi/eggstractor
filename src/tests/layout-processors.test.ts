import { collectTokens } from '../services';
import { transformToScss } from '../transformers';
import { createTestData } from '../utils/test.utils';
import testDataAlignment from './fixtures/figma-test-data_layout-alignment.json';
import testDataDirection from './fixtures/figma-test-data_layout-direction.json';
import testDataWidth from './fixtures/figma-test-data_width.json';
import testDataHeight from './fixtures/figma-test-data_height.json';

describe('Layout Processors', () => {
  it('should process layout alignment correctly', async () => {
    const { setupTest } = createTestData(testDataAlignment);
    const testSetup = await setupTest();

    global.figma = testSetup.figma;

    const tokens = await collectTokens(jest.fn());
    const { result } = transformToScss(tokens);
    expect(result).toMatchSnapshot('alignment');
  });

  it('should process layout direction correctly', async () => {
    const { setupTest } = createTestData(testDataDirection);
    const testSetup = await setupTest();

    global.figma = testSetup.figma;

    const tokens = await collectTokens(jest.fn());
    const { result } = transformToScss(tokens);
    expect(result).toMatchSnapshot('direction');
  });

  it('should process layout width correctly', async () => {
    const { setupTest } = createTestData(testDataWidth);
    const testSetup = await setupTest();

    global.figma = testSetup.figma;

    const tokens = await collectTokens(jest.fn());
    const { result } = transformToScss(tokens);
    expect(result).toMatchSnapshot('width');
  });

  it('should process layout height correctly', async () => {
    const { setupTest } = createTestData(testDataHeight);
    const testSetup = await setupTest();

    global.figma = testSetup.figma;

    const tokens = await collectTokens(jest.fn());
    const { result } = transformToScss(tokens);
    expect(result).toMatchSnapshot('height');
  });
});
