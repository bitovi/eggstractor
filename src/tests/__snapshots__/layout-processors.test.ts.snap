// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout Processors should process layout alignment correctly: alignment 1`] = `
"// Generated SCSS Variables
$brand-primary: #00464a

// Generated SCSS Mixins
@mixin alignment-bottom-left
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff

@mixin alignment-bottom-left_object-1
 background: $brand-primary

@mixin alignment-bottom-left_object-2
 background: $brand-primary

@mixin alignment-bottom-left_object-3
 background: $brand-primary

@mixin alignment-bottom-center
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin alignment-bottom-center_object-1
 background: $brand-primary

@mixin alignment-bottom-center_object-2
 background: $brand-primary

@mixin alignment-bottom-center_object-3
 background: $brand-primary

@mixin alignment-bottom-right
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: flex-end

@mixin alignment-bottom-right_object-1
 background: $brand-primary

@mixin alignment-bottom-right_object-2
 background: $brand-primary

@mixin alignment-bottom-right_object-3
 background: $brand-primary

@mixin alignment-center-left
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff

@mixin alignment-center-left_object-1
 background: $brand-primary

@mixin alignment-center-left_object-2
 background: $brand-primary

@mixin alignment-center-left_object-3
 background: $brand-primary

@mixin alignment-center-center
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin alignment-center-center_object-1
 background: $brand-primary

@mixin alignment-center-center_object-2
 background: $brand-primary

@mixin alignment-center-center_object-3
 background: $brand-primary

@mixin alignment-center-right
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: flex-end

@mixin alignment-center-right_object-1
 background: $brand-primary

@mixin alignment-center-right_object-2
 background: $brand-primary

@mixin alignment-center-right_object-3
 background: $brand-primary

@mixin alignment-top-left
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff

@mixin alignment-top-left_object-1
 background: $brand-primary

@mixin alignment-top-left_object-2
 background: $brand-primary

@mixin alignment-top-left_object-3
 background: $brand-primary

@mixin alignment-top-center
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin alignment-top-center_object-1
 background: $brand-primary

@mixin alignment-top-center_object-2
 background: $brand-primary

@mixin alignment-top-center_object-3
 background: $brand-primary

@mixin alignment-top-right
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: flex-end

@mixin alignment-top-right_object-1
 background: $brand-primary

@mixin alignment-top-right_object-2
 background: $brand-primary

@mixin alignment-top-right_object-3
 background: $brand-primary

"
`;

exports[`Layout Processors should process layout direction correctly: direction 1`] = `
"// Generated SCSS Variables
$brand-primary: #00464a

// Generated SCSS Mixins
@mixin vertical-layout
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin vertical-layout_object-1
 background: $brand-primary

@mixin vertical-layout_object-2
 background: $brand-primary

@mixin vertical-layout_object-3
 background: $brand-primary

@mixin horizontal-layout
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff

@mixin horizontal-layout_object-1
 background: $brand-primary

@mixin horizontal-layout_object-2
 background: $brand-primary

@mixin horizontal-layout_object-3
 background: $brand-primary

@mixin wrapped-layout
 display: flex
 flex-direction: row
 gap: 1.25rem
 padding: 0rem
 background: #ffffff

@mixin wrapped-layout_object-1
 background: $brand-primary

@mixin wrapped-layout_object-2
 background: $brand-primary

@mixin wrapped-layout_object-3
 background: $brand-primary

"
`;

exports[`Layout Processors should process layout height correctly: height 1`] = `
"// Generated SCSS Variables
$brand-primary: #00464a

// Generated SCSS Mixins
@mixin height-hug
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin height-hug_object-1
 background: $brand-primary

@mixin height-fill-container
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin height-fill-container_object-1
 background: $brand-primary

@mixin height-fixed-height
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin height-fixed-height_object-1
 background: $brand-primary

@mixin min-height-custom
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin min-height-custom_object-1
 background: $brand-primary

@mixin min-height-variable
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin min-height-variable_object-1
 background: $brand-primary

@mixin max-height-custom
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin max-height-custom_object-1
 background: $brand-primary

@mixin max-height-variable
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin max-height-variable_object-1
 background: $brand-primary

"
`;

exports[`Layout Processors should process layout width correctly: width 1`] = `
"// Generated SCSS Variables
$size-48: 3rem
$brand-primary: #00464a
$size-16: 1rem

// Generated SCSS Mixins
@mixin min-width-variable
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center
 min-width: $size-48

@mixin min-width-variable_object-1
 background: $brand-primary
 min-width: $size-16

@mixin max-width-variable
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center
 max-width: $size-48

@mixin max-width-variable_object-1
 background: $brand-primary
 max-width: $size-16

@mixin width-hug
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin width-hug_object-1
 background: $brand-primary

@mixin width-fill-container
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin width-fill-container_object-1
 background: $brand-primary

@mixin width-fixed-width
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin width-fixed-width_object-1
 background: $brand-primary

@mixin min-width-custom
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin min-width-custom_object-1
 background: $brand-primary

@mixin max-width-custom
 display: flex
 flex-direction: column
 gap: 1.25rem
 padding: 0rem
 background: #ffffff
 justify-content: center

@mixin max-width-custom_object-1
 background: $brand-primary

"
`;
